### PR TITLE
Sanity check tests + missing equals/hashCode implementations.

### PR DESCRIPTION
**Pull request by jcjimenez**
Found some potential problems around `equals`/`hashcode` that I'm not sure we're hitting, but could pose a problem around redundant counts.

### CODE REVIEW

File: src/main/scala/com/microsoft/partnercatalyst/fortis/spark/transforms/locations/Tile.scala [hunk 1]
```diff
@@ -4,10 +4,7 @@ import com.microsoft.partnercatalyst.fortis.spark.sinks.cassandra.dto.Place
 
 import scala.collection.mutable
 
-class TileId(var tileId: String) {
-  var row: Int = -1
-  var zoom: Int = -1
-  var column: Int = -1
+case class TileId(var tileId: String) {
```
Comment (jcjimenez) [inline src/main/scala/com/microsoft/partnercatalyst/fortis/spark/transforms/locations/Tile.scala:7]: Converting to a case class takes care of `hashCode` and `equals` which was returning `false` for instances with identical `tileId` values.

File: src/main/scala/com/microsoft/partnercatalyst/fortis/spark/transforms/locations/Tile.scala [hunk 2]
```diff
@@ -30,6 +30,23 @@ class Tile(var tileId: TileId, var latitudeNorth: Double, var latitudeSouth: Dou
...
+  override def toString = s"Tile($tileId, $latitudeNorth, $latitudeSouth, $longitudeWest, $longitudeEast)"
+
+  def canEqual(other: Any): Boolean = other.isInstanceOf[Tile]
+
+  override def equals(other: Any): Boolean = other match {
+    case that: Tile =>
+      (that canEqual this) &&
+        tileId == that.tileId
+    case _ => false
+  }
```
Comment (jcjimenez) [inline src/main/scala/com/microsoft/partnercatalyst/fortis/spark/transforms/locations/Tile.scala:43]: Making sure this returns true for equal values of `tileId` regardless of lat/long values (and precision).

**Comment by erikschlegel** [comment]
LGTM

**Review by erikschlegel — APPROVED** [review]
LGTM

**Pull request merged**